react/features/whiteboard/middleware.ts

```diff
@@ -0,0 +1,222 @@
+                roomKey: collabLink?.roomKey
+            } : {},
+            conference,
+            participantId: localParticipantId,
+            whiteboardId: action.id,
+            isEnabled: !isEnabled
+        });
+
+        break;
+    }
+    case CONFERENCE_JOIN_IN_PROGRESS: {
+        const { conference } = action;
+
+        // @ts-ignore
+        conference.addCommandListener(WHITEBOARD, ({ value, attributes }) => {
+            handleWhiteboardStatus(store, conference, value, attributes);
+        });
+        break;
+    } // TODO: cleanup PARTICIPANT_DISPLAY_NAME_CHANGED
+    case PARTICIPANT_UPDATED: {
+        if (typeof APP !== 'undefined') {
+            const participant = getLocalParticipant(store.getState());
+
+            if (participant
+                && participant.id === action.participant.id
+                && action.participant.name
+                && participant.name !== action.participant.name) {
+                localStorage.setItem(
+                    WHITEBOARD_STORAGE_KEY,
+                    JSON.stringify({ username: action.participant.name })
+                );
+                dispatch(setUsernameStatus(USERNAME_UPDATE_STATUS.NEW));
+            }
+        }
+        break;
+    }
+    case PARTICIPANT_LEFT: {
+        const whiteboardId = getWhiteboardId(state);
+
+        if (action.participant.id === whiteboardId) {
+            dispatch(disableWhiteboard());
+        }
+        break;
+    }
+    }
+
+    return next(action);
+});
+
+/**
+ * Set up state change listener to perform maintenance tasks when the conference
+ * is left or failed, e.g. Close the whiteboard if it's left open.
+ */
+StateListenerRegistry.register(
+
+    // @ts-ignore
+    state => getCurrentConference(state),
+
+    // @ts-ignore
+    (conference, { dispatch }, previousConference): void => {
+        if (previousConference) {
+
+            dispatch(disableWhiteboard());
+        }
+    });
+
+
+/**
+ * Handles the whiteboard state.
+ *
+ * @param {Store} store - The redux store.
+ * @param {JitsiConference} conference - The current conference.
+ * @param {string} whiteboardId - The current whiteboard id.
+ * @param {Object} attributes - The command attributes.
+ * @returns {void}
+ */
+
+// @ts-ignore
+function handleWhiteboardStatus(store: IStore, conference, whiteboardId: string, { isEnabled, roomId, roomKey, from }) {
+    const { dispatch, getState } = store;
+
+    if (isEnabled === 'false') {
+        dispatch(participantLeft(whiteboardId, conference));
+
+        return;
+    }
+
+    const isAlreadyEnabled = isWhiteboardOpen(getState());
+
+    if (isAlreadyEnabled) {
+        return;
+    }
+
+    const username = getLocalParticipant(getState())?.name;
+
+    localStorage.setItem(
+        WHITEBOARD_STORAGE_KEY,
+        JSON.stringify({ username })
+    );
+
+    dispatch(participantJoined({
+        conference,
+        id: whiteboardId,
+        isFakeParticipant: true,
+        isWhiteboard: true,
```

Comment:
Note for the future, I think we are at a point where we should start adding some `fakeParticipantType` instead.